pandas/core/series.py

```diff
@@ -3272,6 +3272,11 @@ def _try_cast(arr, take_fast_path):
     # This is to prevent mixed-type Series getting all casted to
     # NumPy string type, e.g. NaN --> '-1#IND'.
     if issubclass(subarr.dtype.type, compat.string_types):
+        # GH 16605
+        # If not empty convert the data to dtype
+        if not isna(data).all():
+            data = np.array(data, dtype=dtype)
+
```

Comment:
u can set copy =False i think here
and just use .astype() which copies

Reply:
A lot of tests started failing when I replaced
```
data = np.array(data, dtype=dtype)
```
with
```
data = np.array(data, copy=False).astype(dtype)
```
Do you know why? Is it related to the fact that ``dtype=`` is used only for upcasting but ``.astype(dtype)`` is used for downcasting?

Reply:
yeah this is numpy stringifying things.  I think this might be ok if you follow this with ``copy=False``, IOW this will copy things, so don't need to copy again.